app/controllers/data_explorer/query_controller.rb

```diff
@@ -0,0 +1,236 @@
+class DataExplorer::QueryController < ::ApplicationController
+  requires_plugin DataExplorer.plugin_name
+
+  before_action :check_enabled
+  before_action :set_group, only: %i(group_reports_index group_reports_show group_reports_run)
+  before_action :set_query, only: %i(group_reports_show group_reports_run)
+
+  attr_reader :group, :query
+
+  def check_enabled
+    raise Discourse::NotFound unless SiteSetting.data_explorer_enabled?
+  end
+
+  def set_group
+    @group = Group.find_by(name: params["group_name"])
+  end
+
+  def set_query
+    @query = DataExplorer::Query.find(params[:id])
+  end
+
+  def index
+    queries = DataExplorer::Query.where(hidden: false).order(:last_run_at, :name).to_a
+
+    database_queries_ids = DataExplorer::Query.pluck(:id)
+    Queries.default.each do |params|
+      attributes = params.last
+      query = DataExplorer::Query.new
+      query.id = attributes["id"]
+      query.sql = attributes["sql"]
+      query.name = attributes["name"]
+      query.description = attributes["description"]
+      query.user_id = Discourse::SYSTEM_USER_ID.to_s
+      queries << query unless database_queries_ids.include?(query.id)
+    end
+
+    render_serialized queries, DataExplorer::QuerySerializer, root: 'queries'
+  end
+
+  skip_before_action :check_xhr, only: [:show]
+  def show
+    check_xhr unless params[:export]
+
+    query = DataExplorer::Query.find(params[:id])
+
+    if params[:export]
+      response.headers['Content-Disposition'] = "attachment; filename=#{query.slug}.dcquery.json"
+      response.sending_file = true
+    end
+
+    return raise Discourse::NotFound if !guardian.user_can_access_query?(query) || query.hidden
+    render_serialized query, DataExplorer::QuerySerializer, root: 'query'
+  end
+
+  def groups
+    render_serialized(Group.all, BasicGroupSerializer)
```

Comment:
Just curious, how is this being used because performance might be bad if a site has alot of groups. Also, do we need to filter out automatic groups or implement some sort of access control here?

Reply:
Queries can be assigned to specific group members:
![image](https://user-images.githubusercontent.com/72780/91237077-fd4f1200-e77c-11ea-9caa-ab0ba8909eca.png)

And then you have reports tab under specific group which allows you to evaluate queries:
![image](https://user-images.githubusercontent.com/72780/91237100-122ba580-e77d-11ea-91d8-75bf078990ef.png)

Reply:
Actually I will just select :id and :name instead of having whole BasicGroupSerializer